gitpod-io/gitpod
# Track Git commands in analytics

## Description


## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/6896

## How to test
1. Open a repo in the preview environment: https://track-git-commands.staging.gitpod-dev.com/#https://github.com/gitpod-io/template-python-flask
2. Open [Segment](https://app.segment.com/gitpod/sources/staging_untrusted/debugger)
3. In the Gitpod workspace, open a terminal and execute a command requiring the cred manager, for instance `git clone https://github.com/filiptronicek/test /tmp/repo`- note that this will error out, but that doesn't matter, it should still register in the debugger
4. View the events in Segment, you should see a `git_command` there somewhere (if none is present, wait a couple of seconds)
4. Profit 💵 

## Release Notes

```release-note
Adds analytic tracking to git commands requiring the git credential manager (remote-facing commands), such as:
- `git clone` (for private repositories)
- `git fetch` (for private repositories)
- `git pull` (for private repositories)
- `git push` (for all repositories)
```

## Documentation

/werft analytics=segment|TEZnsG4QbLSxLfHfNieLYGF4cDwyFWoe
